src/components/hack-details.js

```diff
@@ -71,66 +110,142 @@ export function HackDetails(props) {
+              <Grid item xs={10}>
+                <Typography variant="h6">Idea:</Typography>
+              </Grid>
+              <Grid item xs={10}>
+                <HackField
+                  onChange={handleOnChange}
+                  edit={edit}
+                  name={"description"}
+                  displayText={displayData.description}
+                  variant={"body1"}
+                />
+              </Grid>
+              <Grid item xs={10}>
+                <Typography variant="h6">Goal:</Typography>
+              </Grid>
+              <Grid item xs={10}>
+                <HackField
+                  onChange={handleOnChange}
+                  edit={edit}
+                  name={"goal"}
+                  displayText={displayData.goal}
+                  variant={"body1"}
+                />
+              </Grid>
+              {edit ? (
+                <Grid
+                  container
+                  justify="flex-start"
+                  alignItems="center"
+                  className={classes.marginFix}
+                  spacing={1}
+                >
+                  <Grid item>
+                    <Button
+                      onClick={onSubmitUpdate}
+                      color="primary"
+                      variant="outlined"
+                    >
+                      Update
+                    </Button>
+                  </Grid>
+                  <Grid item>
+                    <Button
+                      onClick={e => {
+                        setEdit(false);
+                      }}
+                      color="secondary"
+                      variant="outlined"
+                    >
+                      Cancel
+                    </Button>
+                  </Grid>
+                </Grid>
+              ) : null}
+              {displayData.team !== undefined ? (
+                <Grid item xs={10} className={classes.marginTop}>
+                  <Typography variant="h6">Team Members:</Typography>
+                </Grid>
+              ) : null}
+              <Grid item xs={10}>
+                <Team team={displayData.team} />
               </Grid>
-            ) : null}
-
-            <Grid item xs={10}>
-              <Team team={data.team} />
             </Grid>
-          </Grid>
+          </form>
         </Paper>
       </Grid>
     );
-  } else {
-    return (
-      <Grid
-        data-testid="main-container"
-        container
-        direction="column"
-        justify="center"
-        alignItems="stretch"
-        alignContent="center"
-        className={classes.loading}
-      >
-        <CircularProgress />
-      </Grid>
-    );
   }
+  return (
+    <Grid
+      data-testid="main-container"
+      container
+      direction="column"
+      justify="center"
+      alignItems="stretch"
+      alignContent="center"
+      className={classes.loading}
+    >
+      <CircularProgress />
+    </Grid>
+  );
 }
 
 const mapState = state => ({
```

Comment:
This component is HUGE! 

Can we think about how to split this up our pages into smaller components to make it easier to parse (because im struggling right now).

Reply:
..... sorry.
I've broken it up a little bit which has made it smaller, but It would be good to discuss ways to further break up and shrink the component size